src/app/application.cpp

```diff
@@ -93,25 +94,40 @@ namespace
 
     const QString LOG_FOLDER("logs");
     const char PARAMS_SEPARATOR[] = "|";
+
+    const QString DEFAULT_PORTABLE_MODE_PROFILE_DIR = QLatin1String("profile");
 }
 
 Application::Application(const QString &id, int &argc, char **argv)
     : BaseApplication(id, argc, argv)
     , m_running(false)
     , m_shutdownAct(ShutdownDialogAction::Exit)
+    , m_commandLineArgs(parseCommandLine(this->arguments()))
 {
+    setApplicationName("qBittorrent");
+    validateCommandLineParameters();
+
+    QString profileDir = m_commandLineArgs.portableMode
+        ? QDir(QCoreApplication::applicationDirPath()).absoluteFilePath(DEFAULT_PORTABLE_MODE_PROFILE_DIR)
+        : m_commandLineArgs.profileDir;
+
+    Profile::initialize(profileDir, m_commandLineArgs.configurationName);
+
     Logger::initInstance();
     SettingsStorage::initInstance();
     Preferences::initInstance();
 
+    if (m_commandLineArgs.webUiPort > 0) {
```

Comment:
Oops. You're correct. I forgot that this might be unset. In this case this check can stay as it is.